TaskHouseUnitTests/WorkersControllerUnitTests.cs

```diff
@@ -0,0 +1,169 @@
+using System;
+using Xunit;
+using TaskHouseApi.Controllers;
+using TaskHouseApi.Repositories;
+using Microsoft.AspNetCore.Mvc;
+using TaskHouseApi.Model;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+
+
+namespace TaskHouseUnitTests
+{
+    public class WorkersControllerUnitTests
+    {
+        //var
+        WorkersController controller;
+        IWorkerRepository repo;
+
+        //Constructor
+      public WorkersControllerUnitTests()
+        {
+            repo = new FakeWorkerRepository();
+            controller = new WorkersController(repo);
+        }
+
+       
+        [Fact]
+        public async System.Threading.Tasks.Task Get_Worker_with_Id_UnitTest() 
+        { 
+           
+           Worker TestWorker =  new Worker() 
+                { 
+                    Id = 1, 
+                    Username = "1234", 
+                    Password = "+z490sXHo5u0qsSaxbBqEk9KsJtGqNhD8I8mVBdDJls=", //1234
+                    Email = "test@test.com", 
+                    FirstName = "Bob1", 
+                    LastName = "Bobsen1", 
+                    Salt = "upYKQSsrlub5JAID61/6pA=="
+                };
+
+
+           
+            var result = await controller.Get(TestWorker.Id);
+            var resultObjectResult = await controller.Get(TestWorker.Id) as ObjectResult;
+            var resultObject = resultObjectResult.Value as TaskHouseApi.Model.Worker;
+
+           
+            Assert.IsType<ObjectResult>(result);
+            Assert.Equal(TestWorker.Id, resultObject.Id);
+
+        }
+  
+        [Fact]
+        public async System.Threading.Tasks.Task Get_Worker_All_UnitTest() 
+        {
+             repo = new FakeWorkerRepository();
+            
+            IEnumerable<Worker> result = await controller.GetAll();
+
+           
+            Assert.Equal(3, result.Count());
+
+        }
+
+      
+        [Fact]
+        public async System.Threading.Tasks.Task Create_Worker_UnitTest() 
+        { 
+            
+           Worker TestWorker =  new Worker() 
+                { 
+                   Id = 5, 
+                   Username = "Tusername", 
+                   Password = "+z490sXHo5u0qsSaxbBqEk9KsJtGqNhD8I8mVBdDJls=", //1234
+                   Email = "test@test.com", 
+                   FirstName = "Bob7", 
+                   LastName = "Bobsen6", 
+                   Salt = "upYKQSsrlub5JAID61/6pA=="
+                };
+
+
+             
+            await controller.Create(TestWorker);
+            var result = await controller.Get(TestWorker.Id);
+            var resultObjectResult = await controller.Get(TestWorker.Id) as ObjectResult;
+            var resultObject = resultObjectResult.Value as TaskHouseApi.Model.Worker;
+
+          
+            Assert.IsType<ObjectResult>(result);
+            Assert.Equal(TestWorker.Id, resultObject.Id);
+
+        }
+
+
+
+   
+        
+        [Fact]
+        public async System.Threading.Tasks.Task Delete_Worker_UnitTest() 
```

Comment:
No need for Unittest in method name, as the testing is happening in a seperate project from the rerst of the code. Should be removed thorugh the entire file